packages/dev-server/src/dev-server.ts

```diff
@@ -6,6 +6,53 @@ import type http from 'http'
 import path from 'path'
 import type { Env, Fetch, EnvFunc, Adapter, LoadModule } from './types.js'
 
+/**
+ * Removes the basePath prefix from a route string.
+ *
+ * @param basePath - The base path to remove.
+ * @param route - The route string to process.
+ * @returns The route string with basePath removed from the start.
+ *
+ * @example
+ * ```ts
+ * removeBasePath("/docs/", "/docs/foo/bar") -> "/foo/bar"
+ * removeBasePath("/docs", "/docs/foo/bar") -> "/foo/bar"
+ * ```
+ */
+export const removeBasePath = (basePath: string, route: string): string => {
```

Comment:
This is a utility function used only internally, so I don't think it needs to be exported. Unit tests are unnecessary and reliability can be ensured with E2E tests only.

```suggestion
const removeBasePath = (basePath: string, route: string): string => {
```